[Windows ARM64] Unhandled exception when running dotnet test

## Steps to reproduce

Using the latest arm64 .NET 5.0 preview SDK:

```
> .\dotnet-install.ps1 -Architecture arm64 -Version 5.0.100-preview.6.20318.15
> dotnet new xunit
> dotnet test
```

## Expected behavior

Should successfully run the test included in the template.

## Actual behavior

Throws an exception:
```
A total of 1 test files matched the specified pattern.
Failed to launch testhost with error: System.AggregateException: One or more errors occurred. (The specified executable is not a valid application for this OS platform.)
 ---> System.ComponentModel.Win32Exception (216): The specified executable is not a valid application for this OS platform.
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.ProcessHelper.LaunchProcess(String processPath, String arguments, String workingDirectory, IDictionary`2 envVariables, Action`2 errorCallback, Action`1 exitCallBack, Action`2 outputCallBack)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DotnetTestHostManager.LaunchHost(TestProcessStartInfo testHostStartInfo, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DotnetTestHostManager.<>c__DisplayClass38_0.<LaunchTestHostAsync>b__0()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__276_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DotnetTestHostManager.LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources, String runSettings)

Test Run Aborted.
C:\Users\eitsarpa\AppData\Local\Microsoft\dotnet\sdk\5.0.100-preview.6.20318.15\Microsoft.TestPlatform.targets(32,5): error MSB4181: The "Microsoft.TestPlatform.Build.Tasks.VSTestTask" task returned false but did not log an error. [C:\Users\eitsarpa\devel\tests\msbuild-repro\msbuild-repro.csproj]
```

## Diagnostic logs

[logs.zip](https://github.com/microsoft/vstest/files/4865847/logs.zip)

## Additional Info

Reproduces with the NUnit template as well. Works without issues using the x86 sdk.

## Environment

vstest.console version: Microsoft (R) Test Execution Command Line Tool Version 16.7.0
OS info: Windows 10 build 18363 on Surface Pro X

--- a/src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs
+++ b/src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs
@@ -230,7 +230,7 @@ namespace Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting
 
             // If testhost.exe is available use it, unless user specified path to dotnet.exe, then we will use the testhost.dll
             bool testHostExeFound = false;
-            if (!useCustomDotnetHostpath && this.platformEnvironment.OperatingSystem.Equals(PlatformOperatingSystem.Windows))
+            if (!useCustomDotnetHostpath && this.platformEnvironment.OperatingSystem.Equals(PlatformOperatingSystem.Windows) && (this.platformEnvironment.Architecture == PlatformArchitecture.X64 || this.platformEnvironment.Architecture == PlatformArchitecture.X86))
             {
                 var exeName = this.architecture == Architecture.X86 ? "testhost.x86.exe" : "testhost.exe";
                 var fullExePath = Path.Combine(sourceDirectory, exeName);

--- a/src/vstest.console/Processors/EnableBlameArgumentProcessor.cs
+++ b/src/vstest.console/Processors/EnableBlameArgumentProcessor.cs
@@ -330,9 +330,7 @@ namespace Microsoft.VisualStudio.TestPlatform.CommandLine.Processors
         private bool IsHangDumpCollectionSupported()
         {
             var dumpCollectionSupported =
-                this.environment.OperatingSystem != PlatformOperatingSystem.OSX
-                && this.environment.Architecture != PlatformArchitecture.ARM64
-                && this.environment.Architecture != PlatformArchitecture.ARM;
+                this.environment.OperatingSystem != PlatformOperatingSystem.OSX;
 
             if (!dumpCollectionSupported)
             {

